File: src/main/java/ac/nsbm/onvent/service/UserService.java
package ac.nsbm.onvent.service;

import ac.nsbm.onvent.model.entity.User;
import ac.nsbm.onvent.repository.UserRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class UserService {

    private final UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public User createUser(User user) {
        // Check if user with the same email already exists
        if (userRepository.findByEmail(user.getEmail()).isPresent()) {
            throw new RuntimeException("User with email " + user.getEmail() + " already exists");
        }
        return userRepository.save(user);
    }

    public List<User> getAllUsers() {
        return userRepository.findAll();
    }

    public Optional<User> getUserById(Long id) {
        return userRepository.findById(id);
    }

    public User updateUser(Long id, User userDetails) {
        User user = userRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("User not found with id: " + id));

        user.setName(userDetails.getName());
        user.setEmail(userDetails.getEmail());
        user.setPassword(userDetails.getPassword());

        return userRepository.save(user);
    }

    public void deleteUserById(Long id) {
        User user = userRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("User not found with id: " + id));
        
        userRepository.delete(user);
    }
}

File: frontend/onvent-frontend/src/services/userService.js
import api from './api';

class UserService {
  // Create a new user
  createUser(user) {
    return api.post('/users/create', user);
  }

  // Get all users
  getAllUsers() {
    return api.get('/users/all');
  }

  // Get user by ID
  getUserById(id) {
    return api.get(`/users/${id}`);
  }

  // Update user by ID
  updateUser(id, user) {
    return api.put(`/users/update/${id}`, user);
  }

  // Delete user by ID
  deleteUser(id) {
    return api.delete(`/users/delete/${id}`);
  }
}

export default new UserService();